packages/discovery-react-components/src/utils/document/documentUtils.ts

```diff
@@ -84,7 +88,26 @@ export function getTextNodeAndOffset(node: Node, offset: number): NodeOffset {
     throw new Error(`Failed to find text node. Node: ${node.textContent}, offset: ${offset}`);
   }
 
-  const textOffset = Math.max(0, offset - runningOffset);
+  let textOffset = Math.max(0, offset - runningOffset);
+
+  // If attribute 'original-text' is present then
+  // the string contains some encoded entities
+  // so, we need adjust the text offset
+  const originalText = nodeElement.getAttribute('data-orig-text');
```

Comment:
For consistency with the other code, you could reference this as `nodeElement.dataset.origText` (see `dataset.childBegin` above).